File: Python/ps_baekjoon/1062/main.py
import sys
from collections import defaultdict

readline, write = (sys.stdin.readline, sys.stdout.write)
exist = ["a", "n", "t", "i", "c"]
vst = defaultdict(bool)
res = 0

def count():
    cnt = 0
    for word in words:
        if all(map(lambda x: vst[x], word)):
            cnt += 1
    return cnt


def solve(cnt=5, init='a'):
    global res
    if cnt >= k:
        res = max(res, count())
        return
    for i in range(ord(init), ord("z") + 1):
        i = chr(i)
        if vst[i]:
            continue
        vst[i] = True
        solve(cnt + 1, chr(ord(init)+1))
        vst[i] = False
        
n, k = [int(i) for i in readline().split()]
words = [set(readline().rstrip()) for _ in range(n)]
for e in exist:
    vst[e] = True
solve()
write("{0}".format(res))

File: C++/ps_baekjoon/1062/main.cpp
#include <bits/stdc++.h>
using namespace std;

constexpr array<char, 5> exist{'a', 'n', 't', 'i', 'c'};
int n, k, res;
vector<string> words;
string input;
bool vst['z' + 1];

int count()
{
    int cnt = 0;
    for (const auto& word : words)
    {
        bool flag = true;
        for (const auto& e : word)
            if (!vst[e])
            {
                flag = false;
                break;
            }
        if (flag)
            ++cnt;
    }
    return cnt;
}

void solve(int cnt = 0, int idx = 'a')
{
    if (cnt == k)
    {
        res = max(res, count());
        return;
    }
    for (int i = idx; i <= 'z'; ++i)
    {
        if (vst[i])
            continue;
        vst[i] = true;
        solve(cnt + 1, i);
        vst[i] = false;
    }
}

int main()
{
    cin.tie(0)->sync_with_stdio(0);
    cin >> n >> k;
    k -= 5;
    words.reserve(n);
    while (n--)
    {
        cin >> input;
        words.push_back(input);
    }
    for (const auto& e : exist)
        vst[e] = true;
    solve();
    cout << res;
    return 0;
}